modules/kubernetes/fluxcd-v2/main.tf

```diff
@@ -0,0 +1,205 @@
+/**
+ * # Flux V2
+ *
+ * Installs and configures [flux2](https://github.com/fluxcd/flux2) with Azure DevOps.
+ *
+ * The module is meant to offer a full bootstrap and confiugration of a Kubernetes cluster
+ * with Fluxv2. A "root" repository is created for the bootstrap configuration along with a
+ * repository per namepsace passed in the variables. The root repository will receive `cluster-admin`
+ * permissions in the cluster while each of the namespace repositories will be limited to their
+ * repsective namespace. The CRDs, component deployments and bootstrap configuration are both
+ * added to the Kubernetes cluster and commited to the root repository. While the namespace
+ * configuration is only comitted to the root repository and expected to be reconciled through
+ * the bootstrap configuration.
+ *
+ * <p align="center"><img src="../../../assets/fluxcd-v2.jpg"></p>
+ */
+
+terraform {
+  required_version = "0.13.5"
+
+  required_providers {
+    flux = {
+      source  = "fluxcd/flux"
+      version = "0.0.4"
+    }
+    azuredevops = {
+      source  = "xenitab/azuredevops"
+      version = "0.2.1"
+    }
+    kubernetes = {
+      source  = "hashicorp/kubernetes"
+      version = "1.13.3"
+    }
+    kubectl = {
+      source  = "gavinbunney/kubectl"
+      version = "1.9.1"
+    }
+  }
+}
+
+locals {
+  repo_url = "http://git-cache-http-server/dev.azure.com/${var.azdo_org}/${var.azdo_proj}/_git/${var.azdo_repo}"
+}
+
+# FluxCD
+data "flux_install" "main" {
+  target_path = var.git_path
+}
+
+data "flux_sync" "main" {
+  url         = local.repo_url
+  target_path = var.git_path
+  branch      = var.branch
+  interval    = 60000000000
+}
+
+data "flux_sync" "groups" {
+  for_each = {
+    for ns in var.namespaces :
+    ns.name => ns
+    if ns.flux.enabled
+  }
+
+  url         = "http://git-cache-http-server/dev.azure.com/${var.azdo_org}/${var.azdo_proj}/_git/${each.key}"
+  branch      = var.branch
+  target_path = var.git_path
+  name        = each.key
+  interval    = 60000000000
+}
+
+# Azure DevOps
+data "azuredevops_project" "this" {
+  name = var.azdo_proj
+}
+
+resource "azuredevops_git_repository" "this" {
+  project_id = data.azuredevops_project.this.id
+  name       = var.azdo_repo
+  initialization {
+    init_type = "Clean"
+  }
+}
+
+resource "azuredevops_git_repository" "groups" {
+  for_each = {
+    for ns in var.namespaces :
+    ns.name => ns
+    if ns.flux.enabled
+  }
+
+  project_id = data.azuredevops_project.this.id
+  name       = each.key
+  initialization {
+    init_type = "Clean"
+  }
+}
```

Comment:
@phillebaba does this create a new repo per group, in the project for XKS?

Reply:
I creates a repo per namespace in the same project.